bundles/org.eclipse.ui/src/org/eclipse/ui/internal/SearchDecoration.java

```diff
@@ -68,14 +73,39 @@ private static String getValidationError(String regex) {
 			return ""; //$NON-NLS-1$
 		} catch (PatternSyntaxException e) {
 			String message = e.getLocalizedMessage();
+			StringBuilder sBuilder = new StringBuilder();
 
-			// Only preserve the first line of the original error message.
 			int i = 0;
 			while (i < message.length() && "\n\r".indexOf(message.charAt(i)) == -1) { //$NON-NLS-1$
 				i++;
 			}
+			int j = i + 2;
+			while (j < message.length() && "\n\r".indexOf(message.charAt(j)) == -1) { //$NON-NLS-1$
+				j++;
+			}
+
+			String firstLine = message.substring(0, i);
+
+			String indexString = firstLine.replaceAll("\\D+", ""); //$NON-NLS-1$ //$NON-NLS-2$
+			int errorIndex = Integer.parseInt(indexString);
+
+			sBuilder.append(firstLine);
+			sBuilder.append(System.lineSeparator());
+
+			String secondLine = message.substring(i + 2, j); // the +2 because of the \n\r
```

Comment:
Linux and Mac use only `\n` so the `+2` won't work. You should work with `String::lastIndexOf` and `System.lineSeparator()` instead.

Reply:
Thanks for the suggestion, with @Wittmaxi s help, I don´t need that anymore, so that is fixed